services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -188,10 +188,9 @@ public Response updateUser(final UserRepresentation rep) {
                 session.getProvider(BruteForceProtector.class).cleanUpPermanentLockout(session, realm, user);
             }
 
-            adminEvent.operation(OperationType.UPDATE).resourcePath(session.getContext().getUri()).representation(rep).success();
-
             if (session.getTransactionManager().isActive()) {
                 session.getTransactionManager().commit();
+                adminEvent.operation(OperationType.UPDATE).resourcePath(session.getContext().getUri()).representation(rep).success();
```

Comment:
Need to move this to below the if statement, as it could be the tx is already committed, which would mean the event wouldn't be sent.